packages/xo-server/src/xo-mixins/pool.js

```diff
@@ -0,0 +1,74 @@
+import { identity, isEmpty } from 'lodash'
+
+export default class {
+  constructor(xo) {
+    this._xo = xo
+  }
+
+  async mergeInto({ sources, target, force }) {
+    const targetHost = this._xo.getObject(target.master)
+    const _sources = sources.map(sourceId =>
+      this._xo.getObject(sourceId, 'pool')
+    )
+    let source
+    let sourceHost
+    for (source of _sources) {
+      sourceHost = this._xo.getObject(source.master)
+      if (sourceHost.productBrand !== targetHost.productBrand) {
+        throw new Error(
+          `a ${sourceHost.productBrand} pool cannot be merged into a ${
+            targetHost.productBrand
+          } pool`
+        )
+      }
+    }
+
+    const hostsPatchesDiff = {}
+    let poolPatchesDiff = []
+    for (source of _sources) {
+      poolPatchesDiff = poolPatchesDiff.concat(
+        this._xo.getPatchesDifference(source.master, target.master)
+      )
+      hostsPatchesDiff[source.id] = this._xo.getPatchesDifference(
+        target.master,
+        source.master
+      )
+    }
```

Comment:
You can use a single loop.

Reply:
I used a single loop to get patches that need to be installed on source or target.